com.unity.ml-agents/Tests/Editor/TrainingAnalyticsSideChannelTests.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+using Google.Protobuf;
+using Unity.MLAgents.Analytics;
+using Unity.MLAgents.SideChannels;
+using Unity.MLAgents.CommunicatorObjects;
+
+
+namespace Unity.MLAgents.Tests
+{
+    public class TrainingAnalyticsSideChannelTests
+    {
+        [Test]
+        public void TestTrainingEnvironmentReceived()
+        {
+            var anyMsg = Google.Protobuf.WellKnownTypes.Any.Pack(new TrainingEnvironmentInitialized());
+            var anyMsgBytes = anyMsg.ToByteArray();
+            var sideChannel = new TrainingAnalyticsSideChannel();
+            using (new AnalyticsUtils.DisableAnalyticsSending())
+            {
+                sideChannel.ProcessMessage(anyMsgBytes);
```

Comment:
not sure I understand the point of this test.

Reply:
Mainly for code coverage, since there's no output to test. Added a docstring at the top of the file indicating this.

We could potential mock the event code to ensure they're being called.